File: MH/DAY2/11727.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs.readFileSync(filePath).toString().trim();
input = +input;
const dp = [0, 1, 3];

for (let i = 3; i <= input; i++) {
  dp[i] = dp[i - 1] + dp[i - 2];
  dp[i - 1] = dp[i];
  if (i % 2 === 0) {
    dp[i] = (dp[i] + 2) % 10007;
  }
  if (i % 2 !== 0) {
    dp[i] = (dp[i] + 1) % 10007;
  }
}

console.log(dp[input]);

File: SY/다시풀기/11727.py
#참고 해결
n=int(input())
dp=[0]*(n+1)
dp[1]=1
if n>1:
	dp[2]=3

	for i in range(3,n+1):
		dp[i]=(dp[i-2]*2)+dp[i-1]
print(dp[n]%10007)